ocdapi/migrations/0002_auto_20180326_1429.py

```diff
@@ -18,23 +20,38 @@ def unmangle_identifier(apps, schema_editor):
 
         added_space = r'^([A-Za-z]+)\s([-\d]+)$'
 
-        chicago_bills = Bill.objects.filter(from_organization__jurisdiction__id='ocd-jurisdiction/country:us/state:il/place:chicago/government').filter(identifier__iregex=added_space)
-
-        for bill in chicago_bills:
-            match = re.match(added_space, bill.identifier)
-            unmangled_identifier = '{mangled_prefix}{count}'.format(mangled_prefix=match.group(1), 
-                                                                    count=match.group(2))
-
-            print('{} becomes {}'.format(bill.identifier, unmangled_identifier))
-            try:
-                duplicate = Bill.objects.get(identifier=unmangled_identifier)
-                print('{} - duplicate found. Deleting.'.format(unmangled_identifier))
-                duplicate.delete()
-            except Bill.DoesNotExist: 
-                pass
-
-            bill.identifier = unmangled_identifier
-            bill.save()
+        with transaction.atomic(), connection.cursor() as cursor:
+            cursor.execute("""
+                DECLARE bills_cursor CURSOR FOR
+                SELECT identifier FROM opencivicdata_bill as bill
+                INNER JOIN opencivicdata_organization as organization
+                ON bill.from_organization_id=organization.id 
+                WHERE organization.jurisdiction_id='ocd-jurisdiction/country:us/state:il/place:chicago/government' 
+                AND bill.identifier ~* '(^([A-Za-z]+)\s([-\d]+)$)'
+            """)
+
+            while True:
+                cursor.execute("FETCH 2 FROM bills_cursor")
```

Comment:
This is going to take a long time, should we increase this size?

Reply:
Of course! I was testing it locally. Good catch.